api/src/main/java/org/apache/iceberg/exceptions/DuplicateWAPCommitException.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.exceptions;
+
+/**
+ * This exception occurs when the WAP workflow detects a duplicate wap commit. This helps clients
+ * to detect duplicate snapshots that are connected by the same wap id.
+ */
+public class DuplicateWAPCommitException extends ValidationException {
```

Comment:
This helps with making WAP workflow idempotent. We use this exception on the DataFrameWriter to differentiate from a generic validation error so that we can NOOP and report success if the wap id was already committed.